src/components/lib/sl_kernel/sl_raw.c

```diff
@@ -333,3 +364,40 @@ sl_thd_free(struct sl_thd *t)
 	sl_thd_free_no_cs(t);
 	sl_cs_exit();
 }
+
+int
```

Comment:
I don't think that we'll find much use for all of this migration complexity. It spans most of the abstraction layers.